tabula/io.py

```diff
@@ -678,10 +678,19 @@ def _extract_from(raw_json, pandas_options=None):
 
         df = pd.DataFrame(data=list_data, columns=_columns, **pandas_options)
 
-        for c in df.columns:
-            df[c] = pd.to_numeric(df[c], errors="ignore")
+        if pandas_options.get('dtype', None) == int:
```

Comment:
I misunderstood your intention.
I think this None value handling for str is redundant because `dtype=str` keeps null value as `np.nan` without modification.
Just adding

```
if not pandas_options.get("dtype"):
```

before `for c in df.columns` would be enough.

Could you check it?